extensions/azure/storage/AzureDataLakeStorage.cpp

```diff
@@ -72,4 +77,62 @@ std::optional<uint64_t> AzureDataLakeStorage::fetchFile(const FetchAzureDataLake
   }
 }
 
+bool AzureDataLakeStorage::matchesPathFilter(const std::string& base_directory, const std::string& path_filter, std::string path) {
```

Comment:
Consider using `std::string_view` instead of `const std::string&` to pass strings.

Clang-Tidy: Method 'matchesPathFilter' can be made static

Reply:
Due to the regex functions' std::string requirements only 1 parameter could be changed to std::string_view. Also moved matchers to anonymous namespace in ef0172d119c472a1c91a10aecbe531a112ade2bf